client/packages/common/src/ui/layout/tables/material-react-table/usePersistDataOnRefetch.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ *
+ */
+
+import { useEffect, useRef } from 'react';
+import { MRT_RowData } from 'material-react-table';
+
+export const usePersistDataOnRefetch = <T extends MRT_RowData>(
```

Comment:
I think the correct way to do this is to lean on tanstack react query instead - can use `keepPreviousData` in the query, and `isFetching` for loading state.

Then shouldn't need any of your other changes:

https://github.com/msupply-foundation/open-msupply/compare/9340-keep-data-on-refetch...9340-keep-data-on-refetch(use-tanstack)